src/program.cpp

```diff
@@ -221,15 +225,44 @@ std::vector<argument> generic_eval(const module& p,
                     assert(results.find(i) != results.end());
                     return results[i];
                 });
-            results.emplace(ins, trace(ins, [&] {
-                                return ins->normalized_operator().compute(
-                                    ctx, ins->get_shape(), values);
-                            }));
+
+            const auto& mod_args = ins->module_inputs();
+            auto module_eval     = [&](module_ref smod,
+                                   const std::unordered_map<std::string, argument>& inputs) {
+                // wrap up parameters for sub_modules
+                const auto& param_names = smod->get_parameter_names();
+                parameter_map m;
+                for(const auto& nm : param_names)
+                {
+                    if(contains(inputs, nm))
+                        m[nm] = inputs.at(nm);
+                    else
+                        MIGRAPHX_THROW("Input " + nm + " parameter of module: \"" + mod->name() +
+                                       "\" not exist!");
+                }
+
+                return generic_eval(smod, ctx, m, results, trace);
```

Comment:
Why cant inputs be passed directly into `generic_eval`? We already check if a parameter is missing.

Reply:
fixed